### ISSUE TITLE
[Bug] Very slow operation of Cumsum in tensorflow 1.6

### ISSUE THREAD

**Issue by yjl9122** [tensorflow/tensorflow#19570]
It seems in tensorflow the tf.cumsum operation is extremely slow on GPU, and takes a huge amount of time (~99% of the total time).

In Pytorch, as expected, the sort operation is the one that takes the most time, cumsum is virtually instant on GPU.

I give an issues on other‘s’ gitHub, we get conclusion that is cumsum is very slow aginst pytorch.
 
we issue is https://github.com/bermanmaxim/LovaszSoftmax/issues/6

code is https://github.com/bermanmaxim/LovaszSoftmax/tree/master/tensorflow

```
def lovasz_grad(gt_sorted):
    """
    Computes gradient of the Lovasz extension w.r.t sorted errors
    See Alg. 1 in paper
    """
    gts = tf.reduce_sum(gt_sorted)
    **intersection = gts - tf.cumsum(gt_sorted)
    union = gts + tf.cumsum(1. - gt_sorted)**
    jaccard = 1. - intersection / union
    jaccard = tf.concat((jaccard[0:1], jaccard[1:] - jaccard[:-1]), 0)
    return jaccard
```

**Comment by yjl9122** [comment]
![20180526173149](https://user-images.githubusercontent.com/18137551/40574769-e04b3f30-610a-11e8-8308-e162e7365a16.png)

**Comment by angersson** [comment]
@yjl9122 Can you provide the extra details about your setup as well as full code to replicate in docker from the [New Issue Template](https://github.com/tensorflow/tensorflow/issues/new)?

**Comment by yjl9122** [comment]
@angersson Any news?

**Comment by angersson** [comment]
Thanks for the update and the code, @yjl9122. @zheng-xq, can you take a look at this?

**Comment by d13g0** [comment]
Having the same issue here with cumsum

**Comment by tensorflowbutler** [comment]
Nagging Assignees @zheng-xq, @angersson: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yjl9122** [comment]
No news @tensorflowbutler @angersson @zheng-xq

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Anjum48** [comment]
Was this ever resolved? I'm also running into performance issues using tf.cumsum on GPU

**Comment by s7ev3n** [comment]
@yjl9122 Have you solved this? I am also facing this issue when using LovaszSoftmax. Thx.

**Comment by ekelsen** [comment]
Should be reasonably fixed now.

**Comment by xusky69** [comment]
As of Tensorflow 2.2 this op is still extremely slow on GPU with tf.complex128 inputs

Splitting into real and imag parts and doing the cumsums separately, then merging to complex again did the trick, as this sped up my forward propagation by a factor of 10.
